### ISSUE TITLE
glowStack() causes executing stack to hide until glow is removed, on iOS/Android

### ISSUE THREAD

**Issue by msintov** [LLK/scratch-blocks#1188]
### Expected Behavior

The executing script should have a yellow outline.

### Actual Behavior

The executing script disappears.

### Steps to Reproduce

We have observed this behavior on iOS and Android for scripts small and large, but in my reproducible case here, this behavior occurs for a very large script in the iOS simulator.

1. Open the vertical playground in iOS simulator (I've tried iOS 9.3 and iOS 10.0. I believe it also repros on iOS 10.3. Perhaps also iOS 11).
2. Paste this xml in the text box for importing: 
[xml_for_long_script.txt](https://github.com/LLK/scratch-blocks/files/1423307/xml_for_long_script.txt)
3. Import the script by tapping "Import from XML"
4. Tap the green flag block, which is the first block of the long script
5. Tap "Stack glow last clicked block" button

Note: If you remove the last block at the very bottom of the stack and then repeat steps 4 and 5, the script will not disappear, but notice that the glow outside is not yellow but instead is the color of the blocks it surrounds (as you can see if you page down through the stack).

### Operating System and Browser

Xcode iOS simulator 9.3 and 10.0 on macOS 10.11.6

**Comment by rachel-fenichel** [comment]
Is the system under serious memory pressure when this happens?

**Comment by msintov** [comment]
@rachel-fenichel No, I don't believe this is a memory issue. For the reproducible steps listed above, the bug can happen even if the simulator was just booted, and my guess is that those steps don't take a lot of memory. Note that if the svg attribute to create the gaussian blur is not applied, then the script does not disappear.

**Comment by xmeow** [comment]
I can reproduce this bug under chrome58, but works fine on chrom63. Looks like browser depends.

**Comment by paulkaplan** [comment]
Yikes, large stacks cause some seriously strange behaviors
1. Confirmed the "disappear on stack glow" with the very large stack on iOS 11 iPad Air simulator 
2. Confirm the "stack glow changes color" if you scroll down the page. Here is a screenshot. Scrolled halfway down, you can see where it goes from yellow to the color of the block
![screen shot 2017-12-20 at 2 08 57 pm](https://user-images.githubusercontent.com/654102/34224132-cf00278c-e58f-11e7-9729-f1f4c09aa839.png)
3. Also, the dragging shadow gets super strange with large stacks of blocks.
![image](https://user-images.githubusercontent.com/654102/34224170-f52d859e-e58f-11e7-9e5d-41fc9eaabefd.png)

Maybe we need to use a simpler SVG filter for mobile browsers? Unfortunately this only seems to happen on large stacks...

**Comment by paulkaplan** [comment]
Oh also I can replicate this on desktop Safari 11 on OS 12.6 (Sierra). So strike this being a "mobile browser thing", I'm seeing it as just a "safari" thing. And @xmeow I wasn't able to replicate this on old versions of Chrome on Mac, were you on android? Can you post screenshots?

**Comment by xmeow** [comment]
I work in a electron binding of version 1.7.10 (chrome 58)
![process_versions](https://user-images.githubusercontent.com/3390845/34239756-827cb7ee-e644-11e7-8a1e-78ee2b9e9e21.png)

And this only happens once custom procedure used.
Try to make a custom procedure block and execute(or apply blocklyStackGlowFilter), and it will disappear. 

![fish_kittenblock](https://user-images.githubusercontent.com/3390845/34239900-82c19f02-e645-11e7-9c4e-3e5c0ac15da2.gif)
![blockdisappear](https://user-images.githubusercontent.com/3390845/34239904-8da44398-e645-11e7-9d0d-f9b5e0c8f94f.png)

But everything works fine if I move to chrome63.
![qq 20171221115646](https://user-images.githubusercontent.com/3390845/34239981-098cae14-e646-11e7-9ea6-62d1b7aba3a8.png)

**Comment by paulkaplan** [comment]
So I think I have isolated the root cause of this. It appears that with very tall SVG elements, safari's SVG `feGaussianBlur` shrinks the source element. You can see this by manually removing everything after the gaussian blur from the stack glow filter. 
![just-blur-filter](https://user-images.githubusercontent.com/654102/34261700-08189696-e638-11e7-9db7-e1e4be578242.gif)
The stack glow is a composite of multiple filters, and because the blur part is shrunk vertically, it causes the "change" from yellow to block color shadow when you scroll down.
Furthermore, at tall enough heights, the result of the blur disappears entirely, which because of the way the final composite is done, causes nothing to show up at all. 

I created a JSfiddle with a simplified filter showing the different behaviors:
![group-of-lines-blurring](https://user-images.githubusercontent.com/654102/34261858-8ed2aed8-e638-11e7-9c0a-5752bb9d39b8.gif)
https://jsfiddle.net/mc7qpsvw/9/

@xmeow is it possible to try that jsfiddle in the old version of chrome as well, to confirm it is the same issue?

**Comment by paulkaplan** [comment]
Also pinging @tmickel since it appears you were the original author of these lines, you might find this interesting.

**Comment by paulkaplan** [comment]
@rachel-fenichel @thisandagain one option would be to not apply the glow filter for stacks above a certain size / for certain browsers. For example, in https://github.com/LLK/scratch-blocks/blob/develop/core/block_svg.js#L226-L232 we could check the BBox height of the svg against some constant we determine through experimentation. We might want to try this approach because gaussian blurs on giant SVG elements can be a huge performance hit, even if they do render correctly.

Another way would be to change the glow to something that doesn't break, or if it isn't the same, provide some kind of graceful fallback in the event of a large stack. However, I tried the other related SVG filters (feMorphology + dilate, for example) and they suffer from the same scaling problem. Anyone have other ideas for implementing that kind of blur?

**Comment by thisandagain** [comment]
I wonder if we removed the blur and simply did a "hard" drop shadow on some platforms and/or instances if the performance and rendering issues could be relieved somewhat.

**Comment by rachel-fenichel** [comment]
There's a definite trade-off here between how nice things look and how fast they move.  I'm personally in favor of going to simpler filters/blurs to improve performance.

If we can find a good heuristic to use, I'm also fine with degrading the appearance on specific platforms.

I don't have numbers on the performance implications of these filters, so that might be a good place to start.

**Comment by msintov** [comment]
On the Cozmo project, we found that both large and small stacks would sometimes disappear when executed due to the glow. So from that experience, I think the solution should be to change the effect for all stacks, not just large ones.

**Comment by xmeow** [comment]
@paulkaplan your jsfiddle works fine in electron 1.7.10~
![fish_kittenblock2](https://user-images.githubusercontent.com/3390845/34282798-2b7ea384-e703-11e7-9ac9-bd4b3c544bad.gif)

**Comment by kyleplo** [comment]
Can recreate in iOS 11 Chrome 63

**Comment by msintov** [comment]
@paulkaplan With the recent stack glow fixes, I am curious if this issue is fixed? Thanks!

**Comment by thisandagain** [comment]
@msintov It *should* be, but we'll need to do a little more testing on Safari to confirm.

**Comment by msintov** [comment]
Our team still reproduces this problem, both on iOS 11 iPads as well as on an iOS 10 simulator, with code from tip.

**Comment by benjiwheeler** [comment]
I ran into this today on an iPad Pro with iOS 13. I was in Bouncy Heroes, a relatively large project. I was editing a very long block stack, then switched to full screen stage, ran the project, and exited full screen. The block stack had disappeared.
